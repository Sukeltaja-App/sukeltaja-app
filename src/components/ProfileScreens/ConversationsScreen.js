import React from 'react'
import { View, FlatList, Text } from 'react-native'
import { ListItem } from 'react-native-elements'
import styles from '../../styles/global'
import { formatDate } from '../../utils/dates'
import { connect } from 'react-redux'

const style = {
  subtitle: {
    fontStyle: 'italic',
    fontSize: 14
  }
}

const ConversationsScreen = (props) => {
  // const { conversations } = props
  //
  // const navigate = (item) => (
  //   props.navigation.navigate('Conversation', { item })
  // )
  //
  // const conversationsSortedByDate = () => (
  //   conversations.sort((a, b) => b.startdate.localeCompare(a.startdate))
  // )
  //
  // const showConversations = () => {
  //   if (conversations.length === 0) {
  //     return (
  //       <View style={styles.centered}>
  //         <Text style={styles.h5}>Ei omia tapahtumia.</Text>
  //       </View>
  //     )
  //   }
  //
  //   return(
  //     <View style={styles.noPadding}>
  //       <FlatList
  //         data={conversationsSortedByDate()}
  //         renderItem={({ item }) => {
  //           const { description, startdate } = item
  //
  //           return (
  //             <ListItem
  //               title={description}
  //               subtitle={formatDate(startdate)}
  //               onPress={() => navigate(item)}
  //               subtitleStyle={style.subtitle}
  //               bottomDivider
  //               chevron
  //             />
  //           )}
  //         }
  //         keyExtractor={item => item.id}
  //       />
  //     </View>
  //   )
  // }
  //
  // return showConversations()

  return(
    <View></View>
  )
}

const mapStateToProps = (state) => ({})

export default connect(
  null,
  null
)(ConversationsScreen)
