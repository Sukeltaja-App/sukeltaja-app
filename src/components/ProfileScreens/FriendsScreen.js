import React from 'react'
import { View } from 'react-native'
import { connect } from 'react-redux'

const FriendsScreen = (props) => {

  return(
    <View></View>
  )
}

//const mapStateToProps = (state) => ({})

export default connect(
  null,
  null
)(FriendsScreen)
