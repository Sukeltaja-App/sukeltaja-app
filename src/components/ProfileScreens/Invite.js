import React from 'react'
import { View, Text } from 'react-native'
import { Button } from 'react-native-elements'
import { connect } from 'react-redux'

import styles from '../../styles/global'
import colors from '../../styles/colors'
import { eventToID } from '../../utils/utilityFunctions'
import { joinOngoingEvent } from '../../reducers/eventReducer'
import { checkMessage } from '../../reducers/messageReducer'

const style = {
  buttonJoin: {
    backgroundColor: colors.green
  },
  buttonRemove: {
    backgroundColor: colors.red
  },
  buttonDivider: {
    height: 30
  }
}

const Invite = (props) => {
  const { joinOngoingEvent, checkMessage, user, navigation } = props
  const { message, parent } = navigation.getParam('invProps')

  console.log(message)
  const join = async () => {
    await joinOngoingEvent(eventToID(message.data))
    await checkMessage(message, user.id, 'accepted')
    parent.updateInvites()
    navigation.navigate('InvitesScreen')
  }

  const removeInvite = async () => {
    await checkMessage(message, user.id, 'rejected')
    parent.updateInvites()
    navigation.navigate('InvitesScreen')
  }

  return (
    <View style={styles.noPadding}>
      <Text style={styles.h5}></Text>
      <Button
        title="Hylkää"
        onPress={() => removeInvite()}
        buttonStyle={style.buttonRemove}
      />
      <View style={style.buttonDivider}/>
      <Button
        title="Liity"
        onPress={() => join()}
        buttonStyle={style.buttonJoin}
      />
      <View style={style.buttonDivider}/>
      <Button
        title="<- Takaisin"
        onPress={() => navigation.navigate('InvitesScreen')}
      />
    </View>
  )
}

const mapStateToProps = (state) => ({
  messages: state.messages,
  user: state.user
})

export default connect(
  mapStateToProps,
  { joinOngoingEvent, checkMessage }
)(Invite)
