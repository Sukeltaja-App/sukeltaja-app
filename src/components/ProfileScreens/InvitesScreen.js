import React from 'react'
import { View, FlatList, Text } from 'react-native'
import { ListItem, Button } from 'react-native-elements'
import styles from '../../styles/global'
import { formatDate } from '../../utils/dates'
import { connect } from 'react-redux'
import { joinOngoingEvent } from '../../reducers/eventReducer'

const style = {
  subtitle: {
    fontStyle: 'italic',
    fontSize: 14
  },
  buttonRow: {
    justifyContent: 'center',
    padding: 10
  },
  buttonDivider: {
    width: 20
  }
}

const InvitesScreen = (props) => {

  const join = (message) => {

  }

  const abandon = (message) => {

  }

  const { messages } = props

  const navigate = (item) => (
    props.navigation.navigate('Invite', { item })
  )

  const messagesSortedByDate = () => (
    messages.sort((a, b) => b.created.localeCompare(a.created))
  )

  const showMessages = () => {
    if (messages.length === 0) {
      return (
        <View style={styles.centered}>
          <Text style={styles.h5}>Ei kutsuja.</Text>
        </View>
      )
    }

    return(
      <View style={styles.noPadding}>
        <FlatList
          data={messagesSortedByDate()}
          renderItem={({ item }) => {
            const { sender, created } = item

            return (
              <ListItem
                title={sender}
                subtitle={formatDate(created)}
                onPress={() => navigate(item)}
                subtitleStyle={style.subtitle}
                bottomDivider
                chevron
              >

              <View style={{ ...styles.row, ...style.buttonRow }}>
                <Button
                  title="Liity"
                  onPress={() => join(item)}
                />
                <View style={style.buttonDivider}/>
                <Button
                  title="Hylkää"
                  onPress={() => abandon(item)}
                />
              </View>

              </ListItem>
            )}
          }
          keyExtractor={item => item.id}
        />
      </View>
    )
  }

  return showMessages()

  return(
    <View></View>
  )
}

const mapStateToProps = (state) => ({
  ongoingEvent: state.ongoingEvent,
  messages: state.messages
})

export default connect(
  mapStateToProps,
  { joinOngoingEvent }
)(InvitesScreen)
