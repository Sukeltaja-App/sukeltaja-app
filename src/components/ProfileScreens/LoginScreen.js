import React from 'react'
import { connect } from 'react-redux'
import { View } from 'react-native'
import { Button } from 'react-native-elements'
import t from 'tcomb-form-native'

import { paddingSides } from '../../styles/global'

import userService from '../../services/users'
import { initializeEvents } from '../../reducers/eventReducer'
import { initializeDives } from '../../reducers/diveReducer'
import { login } from '../../reducers/userReducer'

const { Form } = t.form

const style = {
  container: {
    width: '100%',
    backgroundColor: 'white',
    padding: paddingSides,
    paddingBottom: 50
  },
  buttonDivider: {
    height: 20
  }
}

const options = {
  fields: {
    username: {
      label: 'Käyttäjätunnus:',
      error: 'Käyttäjätunnus ei saa olla tyhjä.'
    },
    password: {
      label: 'Salasana',
      error: 'Salasana ei saa olla tyhjä.',
      secureTextEntry: true
    }
  }
}

class LoginScreen extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      credentials: {
        username: '',
        password: ''
      }
    }
  }

  navigate = (value) => this.props.navigation.navigate(value)

  register = async () => {
    const response = await userService.create(this.state.credentials)

    if(response.data) {
      await this.login()
    } else {
      console.log('Registration failed!')
    }
  }

  login = async () => {
    const { login, initializeEvents, initializeDives } = this.props

    await login(this.state.credentials)
    const { user } = this.props

    if (user) {
      userService.setToken(user.token)

      await initializeEvents()
      await initializeDives()

      this.navigate('UserTab')
    } else {
      console.log('Wrong username or password')
    }
  }

  render() {
    const { credentials } = this.state
    const reference = 'form'

    const User = t.struct({
      username: t.String,
      password: t.String,
    })

    return (
      <View style={style.container}>
        <Form
          ref={reference}
          type={User}
          options={options}
          value={credentials}
          onChange={(credentials) => this.setState({ credentials })}
        />

        <Button
          onPress={this.login}
          title="Kirjaudu"
        />

        <View style={style.buttonDivider}/>

        <Button
          onPress={this.register}
          title="Rekisteröidy"
        />

      </View>
    )
  }
}

const mapStateToProps = (state) => ({ user: state.user })

export default connect(
  mapStateToProps,
  { login, initializeEvents, initializeDives }
)(LoginScreen)
