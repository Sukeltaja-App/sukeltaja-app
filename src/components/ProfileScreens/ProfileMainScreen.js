import React from 'react'
import { connect } from 'react-redux'
import { View } from 'react-native'
import { Avatar, Text, Button } from 'react-native-elements'
import styles from '../../styles/global'
import colors from '../../styles/colors'

import userService from '../../services/users'
import { forgetEvents } from '../../reducers/eventReducer'
import { forgetDives } from '../../reducers/diveReducer'
import { logout } from '../../reducers/userReducer'

export const ProfileMainScreen = (props) => {
  const uri = 'https://upload.wikimedia.org/wikipedia/commons/5/50/USS_Scorpion_sail.jpg'

  const navigate = (value) => props.navigation.navigate(value)

  const logoutButton = () => {
    const { logout, forgetEvents, forgetDives } = props

    forgetEvents()
    forgetDives()
    logout()
    userService.setToken(null)

    navigate('LoginScreen')
  }

  return (
    <View style={styles.centered}>
      <Avatar source={{ uri }} size="xlarge" rounded />
      <Text h2>{props.username}</Text>
      <View style={styles.bottom}>
        <Button
          onPress={() => logoutButton()}
          buttonStyle={{ backgroundColor: colors.red }}
          title="Kirjaudu ulos"
          raised
        />
      </View>
    </View>
  )
}

const mapStateToProps = (state) => ({ username: state.user.username })

export default connect(
  mapStateToProps,
  { logout, forgetEvents, forgetDives }
)(ProfileMainScreen)
