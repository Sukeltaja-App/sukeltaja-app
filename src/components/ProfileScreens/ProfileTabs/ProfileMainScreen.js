import React from 'react'
import { connect } from 'react-redux'
import { View } from 'react-native'
import { Avatar, Text, Button } from 'react-native-elements'

import { logout } from '../../../store'
import { getServerListener } from '../../../ServerListener'
import userService from '../../../services/users'
import styles from '../../../styles/global'
import colors from '../../../styles/colors'

export const ProfileMainScreen = (props) => {
  const uri = 'https://upload.wikimedia.org/wikipedia/commons/5/50/USS_Scorpion_sail.jpg'

  const navigate = (value) => props.navigation.navigate(value)

  const logoutButton = () => {
    const { logout } = props

    const serverListener = getServerListener()

    serverListener.disconnect()

    logout()
    userService.setToken(null)

    navigate('LoginScreen')
  }

  return (
    <View style={styles.centered}>
      <Avatar source={{ uri }} size="xlarge" rounded />
      <Text h2>{props.username}</Text>
      <View style={styles.bottom}>
        <Button
          onPress={() => logoutButton()}
          buttonStyle={{ backgroundColor: colors.red }}
          title="Kirjaudu ulos"
          raised
        />
      </View>
    </View>
  )
}

const mapStateToProps = (state) => ({ username: state.user.username })

export default connect(
  mapStateToProps,
  { logout }
)(ProfileMainScreen)
