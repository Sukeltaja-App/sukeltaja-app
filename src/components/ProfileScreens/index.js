import React from 'react'
import {
  createMaterialTopTabNavigator,
  createStackNavigator,
  createSwitchNavigator,
  MaterialTopTabBar
} from 'react-navigation'
import { Icon } from 'react-native-elements'

import Invite from './ProfileTabs/Invite'
import InvitesScreen from './ProfileTabs/InvitesScreen'
import ProfileScreen from './ProfileTabs/ProfileScreen'
//import SettingsScreen from './ProfileTabs/SettingsScreen'

import LoginScreen from './LoginStack/LoginScreen'
import RegisterScreen from './LoginStack/RegisterScreen'
import ResetScreen from './LoginStack/ResetScreen'

import colors from '../../styles/colors'

const TabBarComponent = (props) => <MaterialTopTabBar { ...props } />

const style = {
  activeTintColor: '#fff',
  labelStyle: {
    fontWeight: 'bold',
    fontSize: 10
  },
  showIcon: true
}

const MessageStack = createStackNavigator({
  InvitesScreen: {
    screen: InvitesScreen
  },
  Invite: {
    screen: Invite
  }
}, {
  headerMode: 'none'
})

const ProfileTabs = createMaterialTopTabNavigator({
  Profile : {
    screen: ProfileScreen,
    navigationOptions: {
      tabBarLabel: 'PROFIILI',
      tabBarOptions: style,
      showIcon: true,
      tabBarIcon: ({ tintColor }) => (
        <Icon name='home' type='feather' color={tintColor} />
      )
    }
  },
  Conversations : {
    screen: MessageStack,
    navigationOptions: {
      tabBarLabel: 'KUTSUT',
      tabBarOptions: style,
      tabBarIcon: ({ tintColor }) => (
        <Icon name='inbox' type='feather' color={tintColor} />
      )
    }
  },
  // Settings : {
  //   screen: SettingsScreen,
  //   navigationOptions: {
  //     tabBarLabel: 'ASETUKSET',
  //     tabBarOptions: style,
  //     tabBarIcon: ({ tintColor }) => (
  //       <Icon name='settings' type='feather' color={tintColor} />
  //     )
  //   }
  // }
}, {
  tabBarComponent: props => {
    return(
      <TabBarComponent
        {...props}
        style={{
          backgroundColor: colors.primary_light,
          paddingTop: 10
        }}
      />
    )
  }
})

const LoginStack = createSwitchNavigator({
  ProfileTabs: {
    screen: ProfileTabs
  },
  LoginScreen: {
    screen: LoginScreen
  },
  ResetScreen: {
    screen: ResetScreen
  },
  RegisterScreen: {
    screen: RegisterScreen
  }
}, {
  headerMode: 'none',
  initialRouteName: 'LoginScreen'
})

export default LoginStack
