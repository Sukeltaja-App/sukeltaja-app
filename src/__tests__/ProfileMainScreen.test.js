import React from 'react'

import renderer from 'react-test-renderer'
import { shallow, mount } from 'enzyme'

import { Button } from 'react-native'

import { ProfileMainScreen } from '../components/ProfileScreens/ProfileMainScreen'

beforeAll(() => {
  wrapper = shallow(<ProfileMainScreen />)
  mounted = mount(<ProfileMainScreen />)
})

describe('<ProfileMainScreen/>', () => {
  it('renders the profile screen snapshot', () => {
    const profileScreen = renderer
      .create(wrapper)
      .toJSON()
    expect(profileScreen).toMatchSnapshot()
  })

  it('pressing the button does nothing', () => {
    const button = mounted.find(Button).first()
    button.simulate('click')
    expect(1).toBe(1)
  })
})
