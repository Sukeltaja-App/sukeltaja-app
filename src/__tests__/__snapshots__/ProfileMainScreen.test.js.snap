// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProfileMainScreen/> renders the profile screen snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#eeeeee",
      "flex": 1,
      "paddingHorizontal": 21,
      "paddingTop": 50,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "transparent",
        "borderRadius": 75,
        "height": 150,
        "width": 150,
      }
    }
    theme={
      Object {
        "colors": Object {
          "disabled": "hsl(208, 8%, 90%)",
          "divider": "#bcbbc1",
          "error": "#ff190c",
          "grey0": "#393e42",
          "grey1": "#43484d",
          "grey2": "#5e6977",
          "grey3": "#86939e",
          "grey4": "#bdc6cf",
          "grey5": "#e1e8ee",
          "greyOutline": "#bbb",
          "platform": Object {
            "android": Object {
              "error": "#f44336",
              "primary": "#2196f3",
              "secondary": "#9C27B0",
              "success": "#4caf50",
              "warning": "#ffeb3b",
            },
            "ios": Object {
              "error": "#ff3b30",
              "primary": "#007aff",
              "secondary": "#5856d6",
              "success": "#4cd964",
              "warning": "#ffcc00",
            },
          },
          "primary": "#2089dc",
          "searchBg": "#303337",
          "secondary": "#8F0CE8",
          "success": "#52c41a",
          "warning": "#faad14",
        },
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#bdbdbd",
          "borderRadius": 75,
          "flex": 1,
          "overflow": "hidden",
          "position": "relative",
        }
      }
    >
      <Image
        onLoadEnd={[Function]}
        source={
          Object {
            "uri": "https://upload.wikimedia.org/wikipedia/commons/5/50/USS_Scorpion_sail.jpg",
          }
        }
        style={
          Object {
            "flex": 1,
            "height": null,
            "width": null,
          }
        }
        theme={
          Object {
            "colors": Object {
              "disabled": "hsl(208, 8%, 90%)",
              "divider": "#bcbbc1",
              "error": "#ff190c",
              "grey0": "#393e42",
              "grey1": "#43484d",
              "grey2": "#5e6977",
              "grey3": "#86939e",
              "grey4": "#bdc6cf",
              "grey5": "#e1e8ee",
              "greyOutline": "#bbb",
              "platform": Object {
                "android": Object {
                  "error": "#f44336",
                  "primary": "#2196f3",
                  "secondary": "#9C27B0",
                  "success": "#4caf50",
                  "warning": "#ffeb3b",
                },
                "ios": Object {
                  "error": "#ff3b30",
                  "primary": "#007aff",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
              },
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#8F0CE8",
              "success": "#52c41a",
              "warning": "#faad14",
            },
          }
        }
      />
      <View
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "opacity": 1,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#bdbdbd",
              "flex": 1,
              "height": null,
              "justifyContent": "center",
              "width": null,
            }
          }
          testID="RNE__Image__placeholder"
        />
      </View>
    </View>
  </View>
  <Text
    style={
      Object {
        "fontSize": 42.5,
      }
    }
    theme={
      Object {
        "colors": Object {
          "disabled": "hsl(208, 8%, 90%)",
          "divider": "#bcbbc1",
          "error": "#ff190c",
          "grey0": "#393e42",
          "grey1": "#43484d",
          "grey2": "#5e6977",
          "grey3": "#86939e",
          "grey4": "#bdc6cf",
          "grey5": "#e1e8ee",
          "greyOutline": "#bbb",
          "platform": Object {
            "android": Object {
              "error": "#f44336",
              "primary": "#2196f3",
              "secondary": "#9C27B0",
              "success": "#4caf50",
              "warning": "#ffeb3b",
            },
            "ios": Object {
              "error": "#ff3b30",
              "primary": "#007aff",
              "secondary": "#5856d6",
              "success": "#4cd964",
              "warning": "#ffcc00",
            },
          },
          "primary": "#2089dc",
          "searchBg": "#303337",
          "secondary": "#8F0CE8",
          "success": "#52c41a",
          "warning": "#faad14",
        },
      }
    }
  />
  <View
    style={
      Object {
        "flex": 1,
        "justifyContent": "flex-end",
        "marginBottom": 50,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#fff",
          "borderRadius": 3,
          "shadowColor": "rgba(0,0,0, .4)",
          "shadowOffset": Object {
            "height": 1,
            "width": 1,
          },
          "shadowOpacity": 1,
          "shadowRadius": 1,
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#DF0000",
              "borderColor": "#2089dc",
              "borderRadius": 3,
              "borderWidth": 0,
              "flexDirection": "row",
              "justifyContent": "center",
              "padding": 8,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 18,
                "paddingBottom": 1,
                "paddingTop": 2,
                "textAlign": "center",
              }
            }
          >
            Kirjaudu ulos
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
